tensorflow/core/platform/cloud/retrying_file_system.cc

```diff
@@ -37,55 +42,81 @@ bool IsRetriable(Status status) {
   }
 }
 
-Status CallWithRetries(const std::function<Status()>& f) {
-  int attempts = 0;
+void WaitBeforeRetry(const int delay_seconds) {
+  std::random_device rd;
+  std::mt19937 gen(rd());
```

Comment:
Please call Rand64() instead:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/lib/random/random.h

It uses a static device, so won't reconstruct every call. Something like `const int64 random_ms = Rand64() % 1000`